### PR TITLE
Support baichuan2 for level0 pipeline

### PR DESCRIPTION
**Pull request by plusbang**
## Description

Background: https://github.com/analytics-zoo/nano/issues/1706#issuecomment-2443913860

image

### 3. Summary of the change 

- Fix embedding when padded idx is not None
- Move embedding and lmhead to common
- Add pipeline input parameter for different model type
- Add baichuan related change (change to int64 node, generate ir and blob, convert)
- Add example

### 4. How to test?
- [x] Application test

### CODE REVIEW

File: python/llm/example/NPU/HF-Transformers-AutoModels/LLM/Pipeline-Models/baichuan2.py [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+        type=str,
+        default="baichuan-inc/Baichuan2-7B-Chat",
+        help="The huggingface repo id for the Baichuan2 model to be downloaded"
+        ", or the path to the huggingface checkpoint folder",
+    )
+    parser.add_argument('--prompt', type=str, default="What is AI?",
+                        help='Prompt to infer')
+    parser.add_argument("--n-predict", type=int, default=32, help="Max tokens to predict")
+    parser.add_argument("--max-context-len", type=int, default=1024)
+    parser.add_argument("--quantization_group_size", type=int, default=0)
```
Comment (rnwang04) [inline python/llm/example/NPU/HF-Transformers-AutoModels/LLM/Pipeline-Models/baichuan2.py:55]: Maybe remove this argument for now as we have not support GW for baichuan2 ?

**Review by rnwang04 — APPROVED** [review]
others LGTM

File: python/llm/example/NPU/HF-Transformers-AutoModels/LLM/Pipeline-Models/baichuan2.py (hunk 1 above)
Comment (plusbang) [reply python/llm/example/NPU/HF-Transformers-AutoModels/LLM/Pipeline-Models/baichuan2.py]: > Maybe remove this argument for now as we have not support GW for baichuan2 ?

Sure, have removed.

**Comment by plusbang** [comment]
Merge it first for other model integration. Please let me know if you have any concern or comment @jason-dai  : )

**Pull request merged**